src/common/common.go

```diff
@@ -306,3 +314,55 @@ func DownloadKubeconfigNoingress(ctx context.Context, dir string, ic inventory_c
 
 	return kubeconfigPath, nil
 }
+
+func ParsePrivateKeyPEM(keyData []byte) (*ecdsa.PrivateKey, error) {
+	var privateKeyPemBlock *pem.Block
+	for {
+		privateKeyPemBlock, keyData = pem.Decode(keyData)
+		if privateKeyPemBlock == nil {
+			break
+		}
+
+		switch privateKeyPemBlock.Type {
+		case ECPrivateKeyBlockType:
+			// ECDSA Private Key in ASN.1 format
+			if key, err := x509.ParseECPrivateKey(privateKeyPemBlock.Bytes); err == nil {
+				return key, nil
+			}
+		}
+	}
+
+	// we read all the PEM blocks and didn't recognize one
+	return nil, fmt.Errorf("data does not contain a valid RSA or ECDSA private key")
+}
+
+func GetPublicKeyBytesFromPrivateKeyPEM(keyData []byte) ([]byte, error) {
+	privateKey, err := ParsePrivateKeyPEM(keyData)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not parse public key from private key PEM")
+	}
+	ecdhPubKey, err := privateKey.PublicKey.ECDH()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not fetch public key from private key")
+	}
+	return ecdhPubKey.Bytes(), nil
+}
+
+func MakeEllipticPrivatePublicKeyPems() ([]byte, []byte, error) {
+	privateKey, err := ecdsa.GenerateKey(elliptic.P256(), cryptorand.Reader)
+	if err != nil {
+		return nil, nil, err
+	}
+	derBytes, err := x509.MarshalECPrivateKey(privateKey)
+	if err != nil {
+		return nil, nil, err
+	}
+	ecdhPubKey, err := privateKey.PublicKey.ECDH()
+	if err != nil {
+		return nil, nil, err
+	}
```

Comment:
ECDH is unrelated, we should do something along the lines of:

```suggestion
	derBytes, err = x509.MarshalPKIXPublicKey(&privateKey.PublicKey)
	if err != nil {
		return nil, nil, err
	}
```